src/key.cpp

```diff
@@ -66,7 +66,7 @@ static int ec_privkey_import_der(const secp256k1_context* ctx, unsigned char *ou
     if (end - privkey < 2 || privkey[0] != 0x04u) {
         return 0;
     }
-    size_t oslen = privkey[1];
+    ptrdiff_t oslen = privkey[1];
```

Comment:
Checked that neither of the values whose type changed can ever be negative, or overflow, given that the range of privkey[x] is 0..255.